### PR TITLE
ci(e2e): Harbor trials on real e2b sandboxes through the full rollout path (manual for now)

### PR DESCRIPTION
**Pull request by nblintao**
## Purpose

The GPU path that the sandbox smoke (#2868) cannot see: launcher wiring → rollout workers → session server + TITO with a real model under the strict gate → terminus-2 driving real e2b sandboxes → reward. Rollout only: everything Harbor-specific runs before the optimizer step, and skipping that step lets the recipe's own model fit on 2 GPUs.

## What it adds

- `tests/e2e/agentic/test_harbor_e2b_rollout.py` — `--debug-rollout-only` with the recipe's model and engine settings (GLM-4.7-Flash, `glm47` TITO family and parsers, 2 GPUs, batch 1×2) on **harbor × e2b × terminus-2 × TB2 `fix-git`** (checkout cloned in `prepare()`; TB2 task dirs are native Harbor tasks with prebuilt images). Passes iff the rollout completes with the TITO strict gate armed and ≥1 Harbor trial ran without an exception.
- New CI label `agentic` (+ comment-command policy entry).

## Behavior change

None: registered `disabled` — CI runners have neither the tailnet route to the sandbox service nor its key. Run manually on a `--tailnet` GPU devbox (runbook in the module docstring). Once runners are provisioned, flipping `disabled` to the `agentic` label is the whole enablement.

## Why rollout only

The earlier shape (one GRPO step with Qwen3-0.6B, the only model that trains on 2 GPUs) needed a strict-gate opt-out for that model's degenerate turns (#3113). No Harbor code runs in the optimizer step, and the recipe's training step is validated by the recipe run recorded in #2837's README, so this PR drops the step, the opt-out flag and its plumbing.

## Status

**Passed 2026-09-09** on a 2×H200 devbox against a self-hosted E2B-compatible service: 2 trials, both **reward 1.0** (9 and 30 terminus turns), TITO strict gate armed and clean (`tito_session_mismatch_rate/v1` and all four per-type rates 0.0). The earlier shape of this PR (one GRPO step with Qwen3-0.6B) passed on 2026-09-04 but only with a strict-gate opt-out for that model's degenerate turns; #3113 has that analysis. The environments.md provider row (AgentENV × Harbor) flips in #2837 on this evidence.

### CODE REVIEW

**Comment by nblintao** [comment]
@claude review

File: tests/e2e/agentic/test_harbor_e2b_rollout.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+
+def check_trials():
+    """The training job finishing is not enough: at least one Harbor trial must
+    have reached its verifier (a reward, no exception)."""
+    trial_dirs = sorted(Path(TRIALS_DIR).glob(f"{SMOKE_TASK}__*"))
+    assert trial_dirs, f"no Harbor trial directories under {TRIALS_DIR}"
+    clean = [d for d in trial_dirs if not (d / "exception.txt").exists()]
+    print(f"harbor trials: {len(trial_dirs)} total, {len(clean)} without exception")
+    assert clean, f"every trial under {TRIALS_DIR} ended in an exception; see the newest exception.txt"
```
[the remark this answers is not in the document]
Comment (nblintao) [reply tests/e2e/agentic/test_harbor_e2b_rollout.py:181]: fixed

File: tests/e2e/agentic/test_harbor_e2b_training.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+def preflight():
+    """Fail fast with instructions instead of failing every trial later."""
+    api_url = os.environ.get("E2B_API_URL", "").strip()
+    if not api_url:
+        sys.exit(
+            "set E2B_API_URL (and E2B_SANDBOX_URL) to the sandbox-service control plane; see the module docstring"
+        )
+    key_file = Path(os.environ.get("E2B_API_KEY_FILE", "~/.config/e2b/api_key")).expanduser()
+    if not os.environ.get("E2B_API_KEY", "").strip() and not key_file.is_file():
+        sys.exit(f"no e2b credential: set E2B_API_KEY or put the key at {key_file}")
```
[the remark this answers is not in the document]
Comment (nblintao) [reply tests/e2e/agentic/test_harbor_e2b_training.py]: fixed

**Comment by Shi-Dong** [comment]
@claude review always

**Review by Shi-Dong — APPROVED** [review]
LGTM!

**Comment by guapisolo** [comment]
codex cmt

Jiajun's context for the original Qwen3-0.6B run: the model is too weak for this agent task. `<|endoftext|>` is its pretraining/padding token, and emitting it during chat reflects insufficient post-training rather than a normal alternative to `<|im_end|>`. He encountered the same behavior while building the initial TITO implementation in March 2026. Larger models can still exhibit it, but much more rarely. He suggest use a stronger model, like GLM-4.6-Flash, or Qwen3.5-35B-A3B as instead.

The strict checker must continue to surface this abnormal output. #3167 records this background beside Qwen3's EOS handling.

**Review by guapisolo — APPROVED** [review]
